docs/rst/enhancements/keys/topic-keys.rst

```diff
@@ -5,49 +5,80 @@ Topic Keys
 
 Topic Keys endow developers with the ability of orchestrating efficient and tailored data distribution,
 facilitating streamlined communication and enhancing system scalability.
-In this exploration, we unravel their significance, working principles, and how to use them.
+This exploration unravels their significance, working principles, and how to use them.
+
+Motivation
+^^^^^^^^^^
+
+Robotic systems and applications, in general, often require modularity in order to divide complex problems into smaller, more manageable parts.
+These sub-systems need to model real world objects, maintaining a state of it (for example, in a robot perception pipeline it could be important to track the pose of an certain object in the environment or retrieving the status of the base controller in a mobile robot).
```

Comment:
```suggestion
These sub-systems need to model real world objects, maintaining a state of it (e.g. in a robot perception pipeline it could be important to track the pose of an certain object in the environment or to retrieve the status of the base controller in a mobile robot).
```